### PR TITLE
Samples: Bluetooth: ignore -EALREADY when opening ipc instance

### LINKED ISSUE #54671: Bluetooth: spurious error when using hci_rpmsg
A `IPC service instance initialization failed` message is printed on the log at boot, but Bluetooth will work fine.

This happens because there is another user of the same ipc instance (usually `ipc0`) used for communication with the Bluetooth host. If that user has already opened it (ie as part of the init sequence, before the app runs), then the next calls will return `-EALREADY`.

This is confusing to users, hence why every other call to `ipc_service_open_instance()` in the zephyr tree rightfully ignores the `-EALREADY` errcode.

**Impact**
Annoyance / confusion

### PR DESCRIPTION
**Pull request by jori-nordic**
This instance might be shared by other users that may have opened it earlier in the boot chain.

Fixes #54671 

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

### CODE REVIEW

**Comment by alwa-nordic** [comment]
@carlocaione I added you as a reviewer because you know the rpmsg API, correct? Could you have a look? Is there a reason to care about the EALREADY error?

File: samples/bluetooth/hci_rpmsg/src/main.c [hunk 1]
```diff
@@ -377,7 +377,7 @@ void main(void)
 
 	/* Initialize IPC service instance and register endpoint. */
 	err = ipc_service_open_instance(hci_ipc_instance);
-	if (err) {
+	if (err < 0 && ret != -EALREADY) {
```
Comment (alwa-nordic) [inline samples/bluetooth/hci_rpmsg/src/main.c:380]:
```suggestion
	if (err < 0 && err != -EALREADY) {
```

**Comment by sarah-read** [comment]
To weigh in on the motivation here, I am working on multiprotocol integration for ANT Wireless on nRF53. This error appears in samples based on hci_rpmsg as a result of ANT and BLE sharing the ipc0 instance. This error is going to cause undue concern for our users, but I'd like to avoid forking and dual maintaining a network core image sample.

**Comment by carlocaione** [comment]
> To weigh in on the motivation here, I am working on multiprotocol integration for ANT Wireless on nRF53. This error appears in samples based on hci_rpmsg as a result of ANT and BLE sharing the ipc0 instance. This error is going to cause undue concern for our users, but I'd like to avoid forking and dual maintaining a network core image sample.

Feel free to push a PR with a proposed change we can discuss on.

**Comment by jori-nordic** [comment]
> > To weigh in on the motivation here, I am working on multiprotocol integration for ANT Wireless on nRF53. This error appears in samples based on hci_rpmsg as a result of ANT and BLE sharing the ipc0 instance. This error is going to cause undue concern for our users, but I'd like to avoid forking and dual maintaining a network core image sample.
> 
> Feel free to push a PR with a proposed change we can discuss on.

this is this PR :)

**Comment by carlocaione** [comment]
> this is this PR :)

I was thinking more of an API change for `ipc_service_open_instance()` or a new `ipc_service_is_open()` function indeed.

**Review by alwa-nordic — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**